cmd/juju/resource/validation.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package resource
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	charmresource "github.com/juju/charm/v10/resource"
+	"github.com/juju/errors"
+	"github.com/juju/juju/cmd/juju/application/utils"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/core/resources"
+	"gopkg.in/yaml.v2"
+	"io"
+	"net/http"
+	"strings"
+)
+
+func ValidateResources(resources map[string]charmresource.Meta) error {
```

Comment:
Because I initially thought we could use some of that in the DeployFromRepository on the controller, so I had them exported from the package. And I think we'll need to run some of these in the cmd/juju/application/deployer/charm.go, which we currently don't (I think, gotta  double check that in the other PR). 

But anyways, I thought this was a nice tidy up for these functions in a separate file. We should probably make the ones we don't use outside of the package private.